src/native/libs/System.Globalization.Native/pal_locale.m

```diff
@@ -781,24 +793,31 @@ int32_t GlobalizationNative_GetLocalesNative(UChar* value, int32_t length)
 {
     @autoreleasepool
     {
-        NSLocale *currentLocale = [NSLocale currentLocale];
-        NSString *localeName = @"";
-
-        if (!currentLocale)
-        {
-            return strdup([localeName UTF8String]);
-        }
-
-        if ([currentLocale.languageCode length] > 0 && [currentLocale.countryCode length] > 0)
+        if (NSLocale.preferredLanguages.count > 0)
```

Comment:
Can this ever be 0? From my OSX machine, it doesn't seem like I can make my Preferred Languages list empty.

Reply:
I don't think so, but better to check.

Reply:
Shouldn't be empty, but I found these kind of issues https://github.com/OneSignal/OneSignal-iOS-SDK/issues/201 , decided to have the check.